cmd/soroban-cli/src/commands/custom.rs

```diff
@@ -0,0 +1,29 @@
+use std::process::Command;
+use which::which;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Plugin not provided. Should be `soroban plugin` for a binary `soroban-plugin`")]
+    MissingSubcommand,
+    #[error(transparent)]
+    IO(#[from] std::io::Error),
+    #[error("Cannot find executable: {0}")]
```

Comment:
Something like the cargo errors here might be nice:
```
$ cargo whatever
error: no such subcommand: `whatever`

     View all installed subcommands with `cargo --list`
```

Then `cargo --list` gives you a nice list of all the known subcommands.

Reply:
Done! Even added a suggestion for closely matched words.